src/V3Inst.cpp

```diff
@@ -475,6 +475,34 @@ class InstDeVisitor final : public VNVisitor {
         }
     }
 
+    void visit(AstArg* nodep) override {
+        if (AstArraySel* const arrselp = VN_CAST(nodep->exprp(), ArraySel)) {
+            // handle single element selection from RHS array
+            if (const AstUnpackArrayDType* const arrp
+                = VN_CAST(arrselp->fromp()->dtypep()->skipRefp(), UnpackArrayDType)) {
+                if (!VN_IS(arrp->subDTypep()->skipRefp(), IfaceRefDType)) return;
+                if (VN_AS(arrp->subDTypep()->skipRefp(), IfaceRefDType)->isVirtual()) return;
+                V3Const::constifyParamsEdit(arrselp->bitp());
+                const AstConst* const constp = VN_CAST(arrselp->bitp(), Const);
+                if (!constp) {
+                    arrselp->bitp()->v3warn(E_UNSUPPORTED,
+                                            "Non-constant index in RHS interface array selection");
+                    return;
+                }
+                const string index = AstNode::encodeNumber(constp->toSInt());
+                const AstVarRef* const varrefp = VN_CAST(arrselp->fromp(), VarRef);
+                UASSERT_OBJ(varrefp, arrselp, "No interface varref under array");
```

Comment:
Might break with 2D array of interfaces, ok if that's unsupported but add a test that the error we get is sane.

Reply:
It seems that 2D arrays of interfaces are caught in V3ParseGrammar, and tested for by t_mod_interface_array3 already, as such I haven't done anything more.